proposals/distribution.md

```diff
@@ -0,0 +1,61 @@
+# Abstract
+
+The Docker registry protocol has become the defacto standard across the container registry world ([https://github.com/docker/distribution/blob/master/docs/spec/index.md](https://github.com/docker/distribution/blob/master/docs/spec/index.md)).
+
+In the OCI, having a solid, common distribution specification with conformance testing will ensure long lasting security and interoperability throughout the container ecosystem.
+
+## Proposal
+
+TL;DR; Move [https://github.com/docker/distribution/tree/master/docs/spec](https://github.com/docker/distribution/tree/master/docs/spec) to [https://github.com/opencontainers/distribution-spec](https://github.com/opencontainers/distribution-spec)
+
+This proposal covers the distribution API spec, and while it does not cover the code for the docker-registry, that implementation is considered the reference implementation. There are other implementations of this protocol, not all are open-source though (Google gcr.io, Amazon ECR, CoreOS Quay, Gitlab registry, JFrog Artifactory registry, Huawei Dockyard, etc).
```

Comment:
Is authentication in the scope of the image distribution?

Reply:
Also, shall we call out image signing (trusted image)?

Reply:
Image signing would be very useful for Singularity containers.

Reply:
@yuwaMSFT2 Unless I'm reading the current spec wrong, authentication is [currently included](https://github.com/docker/distribution/tree/master/docs/spec/auth).

Reply:
> Is authentication in the scope of the image distribution?

The authentication for the registry protocol is just http-based authentication. There is a token authentication specification, which may be in scope, but there a lot of ways to actually implement it that may be context-specific.

> Image signing would be very useful for Singularity containers.

This is not in scope. All resources are content-addressable and can be signed in external systems. Early versions of the specification and implementation had integrated signing, but there were a lot of problems with it. In practice, the enforcement needs to be in the hands of the client.

This doesn't mean the registry couldn't be used to store signature blobs, but that would require more thought. In practice, we've found the best approach is to decouple this.

Reply:
For the token authentication - the fact that it is challenging suggests it's even more important to figure out. I would guess that more rather than fewer would want to use token auth, and perhaps the specification can have a default and then fall back cases for a client to implement that are guaranteed to work for most. Without that, we are in a situation of needing a custom implementation for each one, and that defeats the purpose of having a standard. Maybe we could try, and see how far we get? We can always fall back to sticking with just the basic.

And agreed about the signing, given that it's likely very different! It would be good to come back to this at some future point (and as it's more commonly done and discussed).

Reply:
> There is a token authentication specification, which may be in scope, but there a lot of ways to actually implement it that may be context-specific.

I'm in favor of moving over enough of the auth spec to allow clients to authenticate with Docker's `Bearer` approach without needing to leave the new project's specs.  I've filed #37 against this PR with the changes I think we need for that and more detailed motivation.

Reply:
The issue with this approach is that each auth provider will then have to issue docker-style tokens to interact with the registry. With a more open approach, each registry can choose which providers are integrated. There is also the issue of the access control model: the model used in docker tokens is specific to the way docker implements the registry. Other providers may want to choose to have different token models.

From the perspective of the client, most of this is opaque. They pull an image and authorize the pull through whatever flow makes sense for the context and provide an opaque value for an http header.